### PR TITLE
Add util to send "multipart/form-data" HTTP POST

### PR DESCRIPTION
**Pull request by chanseokoh**
In preparation for #517.

Due to some bug on the internal exception reporting system, now I have to employ a workaround to send a multipart/form-data HTTP POST.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/io/HttpUtilWithServerTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  public void testSendPostMultipart() throws IOException {
+    Map<String, String> parameters = new HashMap<>();
+    parameters.put("dp", "/virtual/some-event-type/some-event-name");
+    parameters.put("exception_info",
+        "Exception in thread \"main\" java.lang.IllegalArgumentException: Wrong argument!\n\tat B.main(B.java:73)");
+    parameters.put("product", "CT4E");
+
+    assertEquals(200, HttpUtil.sendPostMultipart(server.getAddress(), parameters));
+    assertEquals("POST", server.getRequestMethod());
+    assertTrue(server.getRequestHeaders().get("User-Agent").startsWith("gcloud-eclipse-tools/0.1.0."));
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/io/HttpUtilWithServerTest.java]: where's the 0.1.0 coming from? I'd love to change that to 1.0.0 but it's probably too painful

File: plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/io/HttpUtil.java [hunk 2]
```diff
@@ -31,6 +41,37 @@
   public static final int DEFAULT_CONNECT_TIMEOUT_MS = 3000;
   public static final int DEFAULT_READ_TIMEOUT_MS = 3000;
 
+  private static final String MULTIPART_BOUNDARY =
+      "---------------------------45224ee4-f3c1-4b23-8df1-4012f722218c"; // some random UUID
+
+  private static HttpTransport transport = new NetHttpTransport();
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/io/HttpUtil.java]: final

File: plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/io/HttpUtil.java [hunk 3]
```diff
@@ -31,6 +41,37 @@
   public static final int DEFAULT_CONNECT_TIMEOUT_MS = 3000;
   public static final int DEFAULT_READ_TIMEOUT_MS = 3000;
 
+  private static final String MULTIPART_BOUNDARY =
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/io/HttpUtil.java]: Where did this come from? Should we be using the same one every time?

File: plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/io/HttpUtilWithServerTest.java (hunk 1 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util.test/src/com/google/cloud/tools/eclipse/util/io/HttpUtilWithServerTest.java]: #1454. We should think about this very soon.

File: plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/io/HttpUtil.java (hunk 3 above)
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/io/HttpUtil.java]: It's some arbitrary UUID I generated. It's perfectly fine to reuse the same boundary every time, as long as it gets its job done to separate multiple parts (i.e., it's a unique boundary value that never appears in any actual parameter values).

Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/io/HttpUtil.java]: Theoretically, we should ensure parameter values don't have this particular boundary value, but the chance that there will be an overlap is basically 0 in practice, so I think just using this boundary is an acceptable compromise.

**Pull request merged**